### DISCUSSION TITLE
django-framework-exceptions does not work

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [projectdiscovery/nuclei#2409] [Q&A] [answered]
I'm testing nuclei against a random django site, it contains the keyword `DisallowedHost`

```
<title>DisallowedHost at /web/settings/settings.py</title>
```

Then I started nuclei to test the template:

```
# nuclei -u hxxp://xxxx:8081/ -debug -tags django
```

And `django-framework-exceptions` did not work:

```
[WRN] [django-framework-exceptions] Could not execute request for hxxp://xxxxx:8081/: could not send file request: could not find file: stat `hxxp://xxxxx:8081/
```

What's wrong?

**Comment by forgedhallpass** [comment, marked as answer]
Hello @CaledoniaProject,

The `django-framework-exceptions` template is a file template that expects a file input (e.g. the server log file).

If these exceptions are also returned in a HTTP response body, then maybe it would make sense creating an HTTP template for it as well. Would you like to give it a try?

**Reply by CaledoniaProject** [reply]
Of course, how should I do that?
Is there any existing template that can be used as a reference?

**Reply by CaledoniaProject** [reply]
@forgedhallpass

**Reply by forgedhallpass** [reply]
You would basically need to duplicate the current template, move it under the right directory structure within the [`nuclei-templates`](https://github.com/projectdiscovery/nuclei-templates) repository and then change the protocol from `file` to HTTP(`requests`). If you search for `requests` within the templates project, you'll find plenty, but you can also rely on the [templating guide](https://nuclei.projectdiscovery.io/templating-guide/) and the [template examples](https://nuclei.projectdiscovery.io/template-examples/http/). Once it's done, you can contribute it via the [Template Contribution](https://github.com/projectdiscovery/nuclei-templates/issues/new?assignees=&labels=nuclei-template&template=submit-template.md) button or by creating a [PR](https://github.com/projectdiscovery/nuclei-templates/pulls).

If you have any follow-up questions, feel free to DM me on [Discord](https://discord.gg/projectdiscovery).

**Reply by CaledoniaProject** [reply]
@forgedhallpass I could duplicate `django-framework-exceptions`, but is that what I'm supposed to do? The extractors would be exactly the same. I mean, is there any way to share the extractors with `file` and `requests` key?

**Reply by forgedhallpass** [reply]
@CaledoniaProject Theoretically it should be possible to extract the payloads (not the whole extractor(s) though) into a file, and then reference them in both templates, but at this point you're better off with duplicating the template. The file name and template ID should be different though.

I've created a [ticket](https://github.com/projectdiscovery/nuclei/issues/1510) for this already a while back. We'll implement it at some point.

**Reply by CaledoniaProject** [reply]
@forgedhallpass https://github.com/projectdiscovery/nuclei-templates/pull/5079/files

**Comment by 0xAgun** [comment]
DisallowedHost is from django..  in django srmettings.py file you'll find a filed called allwoed hosts there you can tey * on it django doesn't allow any other host rather then listed one.. I guess so
